x/stakeibc/keeper/update_redemption_rates_test.go

```diff
@@ -3,125 +3,111 @@ package keeper_test
-	}
+	return []recordtypes.DepositRecord{toBeStakedDepositRecord, toBeTransferedDepositRecord}
 }
 
 func (s *KeeperTestSuite) TestUpdateRedemptionRatesSuccessful() {
-	balancedDelegation := sdkmath.NewInt(5)
-	undelegatedBal := sdkmath.NewInt(3)
-	justDepositedBal := sdkmath.NewInt(3)
-	stSupply := sdkmath.NewInt(10)
-
-	initialRedemptionRate := sdk.NewDec(1)
-	tc := s.SetupUpdateRedemptionRates(balancedDelegation, undelegatedBal, justDepositedBal, stSupply, initialRedemptionRate)
+	depositRecords := s.SetupUpdateRedemptionRates(UpdateRedemptionRateTestCase{
+		balancedDelegation:    sdkmath.NewInt(5),
+		unbalancedDelegation:  sdkmath.ZeroInt(),
+		undelegatedBal:        sdkmath.NewInt(3),
+		justDepositedBal:      sdkmath.NewInt(3),
+		stSupply:              sdkmath.NewInt(10),
+		initialRedemptionRate: sdk.NewDec(1),
+	})
 
-	// sanity check on inputs (check redemptionRate at genesis is 1)
-	s.Require().Equal(initialRedemptionRate, sdk.NewDec(1), "t0 rr")
-
-	records := tc.allRecords
-	s.App.StakeibcKeeper.UpdateRedemptionRates(s.Ctx, records)
-
-	hz, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx, tc.hostZone.ChainId)
-	s.Require().True(found, "hz found")
-	rrNew := hz.RedemptionRate
+	s.App.StakeibcKeeper.UpdateRedemptionRates(s.Ctx, depositRecords)
 
 	expectedNewRate := sdk.NewDec(5 + 3 + 3).Quo(sdk.NewDec(10))
-	s.Require().Equal(rrNew, expectedNewRate, "rr as expected")
+	s.checkRedemptionRateAfterUpdate(expectedNewRate)
 }
 
 func (s *KeeperTestSuite) TestUpdateRedemptionRatesRandomized() {
 	// run N tests, each with random inputs
+	max := int64(1_000_000_000)
+	tc := UpdateRedemptionRateTestCase{
+		balancedDelegation:    sdkmath.NewInt(max),
+		unbalancedDelegation:  sdkmath.ZeroInt(),
+		undelegatedBal:        sdkmath.NewInt(max),
+		justDepositedBal:      sdkmath.NewInt(max),
+		stSupply:              sdkmath.NewInt(max),
+		initialRedemptionRate: sdk.NewDec(1),
+	}
+	depositRecords := s.SetupUpdateRedemptionRates(tc)
 
-	MAX := "1_000_000_000"
-	balancedDelegation, _ := sdk.NewIntFromString(MAX)
-	undelegatedBal, _ := sdk.NewIntFromString(MAX)
-	justDepositedBal, _ := sdk.NewIntFromString(MAX)
-	stSupply, _ := sdk.NewIntFromString(MAX)
-
-	initialRedemptionRate := sdk.NewDec(1)
-	tc := s.SetupUpdateRedemptionRates(balancedDelegation, undelegatedBal, justDepositedBal, stSupply, initialRedemptionRate)
-
-	// sanity check on inputs (check redemptionRate at genesis is 1)
-	s.Require().Equal(initialRedemptionRate, sdk.NewDec(1), "t0 rr")
-
-	records := tc.allRecords
-	s.App.StakeibcKeeper.UpdateRedemptionRates(s.Ctx, records)
-
-	hz, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx, tc.hostZone.ChainId)
-	s.Require().True(found, "hz found")
-	rrNew := hz.RedemptionRate
+	s.App.StakeibcKeeper.UpdateRedemptionRates(s.Ctx, depositRecords)
 
-	numerator := balancedDelegation.Add(undelegatedBal).Add(justDepositedBal)
-	denominator := stSupply
+	numerator := tc.balancedDelegation.Add(tc.undelegatedBal).Add(tc.justDepositedBal)
```

Comment:
These tests are a bit confusing when you omit terms from the formula eg `unbalancedDelegation`. I assume they're omitted bc they're 0, might just be easier to include them

Reply:
In fact we might reduce potential for human-error in these tests by refactoring `checkRedemptionRateAfterUpdate` to take in the components of the RR calc and calc them internally

Reply:
yeah i was going back and forth on this a bit cause it feels a bit sloppy to just re-do the exact same calculation that's done in the original function. I think the two options are:
* Recalculate everything in a helper function
* Use hard coded values (like [here](https://github.com/Stride-Labs/stride/pull/723/files/b987dfaa14634d747b10a472e6d217e07d48738c#diff-22743c5e09b78fab77ad928aa66d988fea39bf5f81e40fc8ce2632a116486070R167))

Reply:
actually I forgot about #725 - I think it's further clarified there. Lmk if you disagree

Reply:
this clarifies it. resolved!